### PR TITLE
EDX_BLND_CLI-39 Take the Library Content button out the Advanced section to the top default level

### PR DESCRIPTION
**Pull request by UvgenGen**
[EDX_BLND_CLI-39](https://youtrack.raccoongang.com/issue/EDX_BLND_CLI-39) Take the Library Content button out the Advanced section to the top default level

### CODE REVIEW

File: cms/djangoapps/contentstore/views/component.py [hunk 1]
```diff
@@ -383,6 +386,37 @@ def create_support_legend_dict():
                         )
                         categories.add(component)
 
+        # Add library problem types.
+        if category == 'library' and not library:
+            disabled_block_names = [block.name for block in disabled_xblocks()]
+            library_problem_types = [problem_type for problem_type in LIBRARY_PROBLEM_TYPES
+                                      if problem_type['component'] not in disabled_block_names]
+            for library_problem_type in library_problem_types:  # pylint: disable=consider-iterating-dictionary
```
Comment (monteri) [inline cms/djangoapps/contentstore/views/component.py:394]: Redundant `# pylint: disable=consider-iterating-dictionary`

File: cms/djangoapps/contentstore/views/component.py [hunk 2]
```diff
@@ -288,7 +291,7 @@ def create_support_legend_dict():
     component_types = COMPONENT_TYPES[:]
 
     # Libraries do not support discussions and openassessment
```
Comment (dyudyunov) [inline cms/djangoapps/contentstore/views/component.py:293]: ... and other libraries

File: cms/envs/common.py [hunk 3]
```diff
@@ -1981,6 +1981,17 @@
     }
 ]
 
+LIBRARY_PROBLEM_TYPES = [
```
Comment (dyudyunov) [inline cms/envs/common.py:1984]: This is actually not a problem type, rather it could be named `LIBRARY_BLOCK_TYPES` or `LIBRARY_XBLOCK_TYPES`

**Comment by UvgenGen** [comment]
> Do I understand correctly that we will not need to add library blocks to the course's advanced settings after this PR will be merged?
Yes)

File: cms/envs/common.py (hunk 3 above)
Comment (UvgenGen) [reply cms/envs/common.py]: updated

File: cms/djangoapps/contentstore/views/component.py (hunk 2 above)
Comment (UvgenGen) [reply cms/djangoapps/contentstore/views/component.py]: updated

File: cms/djangoapps/contentstore/views/tests/test_item.py [hunk 4]
```diff
@@ -2253,6 +2253,9 @@ def setUp(self):
         XBlockStudioConfiguration.objects.create(name='video', enabled=True, support_level="us")
         # ORA Block has it's own category.
         XBlockStudioConfiguration.objects.create(name='openassessment', enabled=True, support_level="us")
+        # Library Sourced Content has it's own category.
```
[the remark this answers is not in the document]
Comment (UvgenGen) [reply cms/djangoapps/contentstore/views/tests/test_item.py]: updated

File: cms/static/sass/assets/_graphics.scss [hunk 5]
```diff
@@ -52,3 +52,10 @@
   height: ($baseline*3);
   background: url('#{$static-path}/images/large-openassessment-icon.png') center no-repeat;
 }
+
+.large-library-icon {
+  display: inline-block;
+  width: ($baseline*3);
+  height: ($baseline*3);
+  background: url('#{$static-path}/images/large-library-icon.png.png') center no-repeat;
```
[the remark this answers is not in the document]
Comment (UvgenGen) [reply cms/static/sass/assets/_graphics.scss]: updated

File: cms/djangoapps/contentstore/views/component.py (hunk 1 above)
Comment (UvgenGen) [reply cms/djangoapps/contentstore/views/component.py]: updated

**Comment by dyudyunov** [comment]
👍

**Pull request merged**